### DISCUSSION TITLE
Basic Configuration for Logger

### DISCUSSION THREAD

**Discussion by soerenetler** [python-telegram-bot/python-telegram-bot#2850] [Q&A] [answered]
Hi ptb-Community
I am currently trying to configure logging with python telegram bot using a json log formatter (python-json-logger). When looking through the ptb library I found that at some places the basic configuration for the logs is configured. Why is this the case? Wouldn't it be enough to just call `logger = logging.getLogger(__name__)`?

One example can be found in the codesection linke below. 
https://github.com/python-telegram-bot/python-telegram-bot/blob/cf4d3cae01dea241e35dde5116b2b950c1a5b61a/examples/errorhandlerbot.py#L14

Thank you in advance,
Sören

**Comment by Bibo-Joshi** [comment, marked as answer]
Hi.

`logging.basicConfig` tells the root logger what to do with incoming log records. `logger = logging.getLogger(__name__)` just gives you a logger for the current module - it does not tell the `logger` what to do with incoming log records. On the contrary, this is usually used to *send* log records via e.g. `logger.warning('this is the captain speaking')`. In fact this is exactly what PTB does - e.g.

https://github.com/python-telegram-bot/python-telegram-bot/blob/ade1529986f5b6d394a65372d6a27045a70725b2/telegram/ext/dispatcher.py#L195

`logging.basicConfig` is used only in the examples (not in the library itself) so that users will see exceptions happening in their callbacks even if they don't have an error handler registered.

**Comment by soerenetler** [comment]
Thank you a lot for your answer. That helped me and I do understand logging now a little better. Thanks!
